internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go

```diff
@@ -0,0 +1,1100 @@
+	DNSService           bool   `tfschema:"dns_service"`
+	HTTPGatewayPort      int64  `tfschema:"http_gateway_port"`
+	Location             string `tfschema:"location"`
+	Name                 string `tfschema:"name"`
+	Username             string `tfschema:"username"`
+	Password             string `tfschema:"password"`
+	ResourceGroup        string `tfschema:"resource_group_name"`
+
+	Authentication       []Authentication                     `tfschema:"authentication"`
+	CustomFabricSettings []CustomFabricSetting                `tfschema:"custom_fabric_setting"`
+	LBRules              []LBRule                             `tfschema:"lb_rule"`
+	NodeTypes            []NodeType                           `tfschema:"node_type"`
+	Sku                  managedcluster.SkuName               `tfschema:"sku"`
+	Tags                 map[string]interface{}               `tfschema:"tags"`
+	UpgradeWave          managedcluster.ClusterUpgradeCadence `tfschema:"upgrade_wave"`
+}
+
+func (k ClusterResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"backup_restore_service": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+		},
+		"dns_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-z0-9]+(-*[a-z0-9])*$`), "The dns name of the cluster must have lowercase letters, numbers and hyphens. The first character must be a letter and the last character a letter or number"),
+		},
+		"dns_service": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+		},
+		"location": azure.SchemaLocation(),
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(4, 23),
+				validation.StringMatch(regexp.MustCompile(`^[a-z0-9]+(-*[a-z0-9])*$`), "The name of the cluster must have lowercase letters, numbers and hyphens. The first character must be a letter and the last character a letter or number")),
+		},
+		"username": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(1, 15),
+				validation.StringMatch(regexp.MustCompile("^[^\\\\/\"\\[\\]:|<>+=;,?*$]{1,14}$"), "User names cannot contain special characters \\/\"\"[]:|<>+=;,$?*@")),
+		},
+		"password": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(8, 123),
+				validation.StringIsNotWhiteSpace),
+		},
+		"resource_group_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotWhiteSpace,
+		},
+
+		"node_type":      nodeTypeSchema(),
+		"authentication": authSchema(),
+		"custom_fabric_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"parameter": {
```

Comment:
I almost feel like this should be 
```suggestion
		"custom_fabric_parameter": {
			Type:     pluginsdk.TypeList,
			Optional: true,
			Elem: &pluginsdk.Resource{
				Schema: map[string]*pluginsdk.Schema{
					"name": {
```

🤔  - WDYT?